### DISCUSSION TITLE
Inquiry Regarding Automatic Deletion of Non-Existent File History Entries

### DISCUSSION THREAD

**Discussion by ryanlo713** [sumatrapdfreader/sumatrapdf#4646] [Ideas]
I would like to inquire whether it is possible to automatically delete entries from the file history corresponding to files that no longer exist. After downloading and reviewing docs, I relocated them to different folders, which has resulted in records of non-existent files remaining in the file history.

I am seeking a one-click solution or an automatic clearing feature to address this issue. Otherwise, I am concerned about the accumulation of multiple entries with identical names in the file history.

Thank you for your assistance.

![image](https://github.com/user-attachments/assets/fbd9c546-7d9a-434b-8eea-05aec81c4c7f)

**Comment by GitHubRulesOK** [comment]
Historically there is no way to know how a user needs their history culled as files that are currently not there may be required to maintain those settings in the future. Some get disturbed when on USB or other temporary added and removed drives.

The methods are EITHER backup list then delete all and add back a few desired OR delete those no longer required OR export the current tagged files as shown in other Q&A about grouping a set for loading different topics
